File: bot-api/java/src/main/java/dev/robocode/tankroyale/botapi/events/HitByBulletEvent.java
package dev.robocode.tankroyale.botapi.events;

import dev.robocode.tankroyale.botapi.BulletState;

/**
 * Event occurring when a bullet has hit your bot.
 */
@SuppressWarnings("unused")
public final class HitByBulletEvent extends BotEvent {

    // Bullet that hit the bot.
    private final BulletState bullet;

    // Damage inflicted by the bullet.
    private final double damage;

    // Remaining energy level after the bullet hit.
    private final double energy;

    /**
     * Initializes a new instance of the HitByBulletEvent class.
     *
     * @param turnNumber is the turn number when the bullet has hit a bot.
     * @param bullet     is the bullet that hit the bot.
     * @param damage     is the damage inflicted by the bullet.
     * @param energy     is the remaining energy level of the bot that got hit.
     */
    public HitByBulletEvent(int turnNumber, BulletState bullet, double damage, double energy) {
        super(turnNumber);
        this.bullet = bullet;
        this.damage = damage;
        this.energy = energy;
    }

    /**
     * Returns the bullet that hit your bot.
     *
     * @return The bullet that hit your bot.
     */
    public BulletState getBullet() {
        return bullet;
    }

    /**
     * Returns the damage inflicted by the bullet.
     *
     * @return The damage inflicted by the bullet.
     */
    public double getDamage() {
        return damage;
    }

    /**
     * Returns the remaining energy level after the bullet hit.
     *
     * @return The remaining energy level after the bullet hit.
     */
    public double getEnergy() {
        return energy;
    }
}

File: bot-api/python/src/robocode_tank_royale/bot_api/events/hit_by_bullet_event.py
from dataclasses import dataclass

from ..bullet_state import BulletState
from .bot_event import BotEvent


@dataclass(frozen=True, repr=True)
class HitByBulletEvent(BotEvent):
    """
    Represents an event when a bot is hit by a bullet.

    This event is triggered when a bullet hits the bot during a turn in the game.
    It provides information about the bullet, the damage caused, and the bot's
    remaining energy after the hit.

    Attributes:
        bullet (BulletState): The bullet that hit the bot.
        damage (float): The damage inflicted on the bot by the bullet.
        energy (float): The bot's remaining energy after being hit.

    """

    bullet: BulletState
    damage: float
    energy: float
